src/overlays/actors/ovl_Bg_Mori_Bigst/z_bg_mori_bigst.c

```diff
@@ -20,39 +36,219 @@ const ActorInit Bg_Mori_Bigst_InitVars = {
+}
+
+void BgMoriBigst_SetupObjectCheck(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_ObjectCheck);
+}
+
+void BgMoriBigst_ObjectCheck(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex)) {
+        thisx->draw = BgMoriBigst_Draw;
+        if (Flags_GetClear(globalCtx, thisx->room) && (PLAYER->actor.posRot.pos.y > 700.0f)) {
+            if (Flags_GetSwitch(globalCtx, (thisx->params >> 8) & 0x3F)) {
+                BgMoriBigst_FinalNoop(this, globalCtx);
+            } else {
+                BgMoriBigst_SpawnStalfos(this, globalCtx);
+            }
+        } else {
+            BgMoriBigst_InitialNoop(this, globalCtx);
+        }
+    }
+}
+
+void BgMoriBigst_InitialNoop(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, NULL);
+}
+
+void BgMoriBigst_SpawnStalfos(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor* stalfos;
+
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_StalfosWait);
+    Flags_UnsetClear(globalCtx, this->dyna.actor.room);
+    stalfos = Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_EN_TEST, 209.0f, 827.0f,
+                                 -3320.0f, 0, 0, 0, 1);
+    if (stalfos != NULL) {
+        this->dyna.actor.child = NULL;
+        this->dyna.actor.initPosRot.rot.z++;
+    } else {
+        // Second Stalfos failure
+        osSyncPrintf("Warning : 第２スタルフォス発生失敗\n");
+    }
+    Flags_SetClear(globalCtx, this->dyna.actor.room);
+}
+
+void BgMoriBigst_StalfosWait(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->dyna.actor.initPosRot.rot.z == 0) &&
+        ((this->dyna.actor.initPosRot.pos.y - 5.0f) <= PLAYER->actor.posRot.pos.y)) {
+        BgMoriBigst_SetupFall(this, globalCtx);
+        func_800800F8(globalCtx, 0xC94, 0x48, &this->dyna.actor, 0);
+    }
+}
+
+void BgMoriBigst_SetupFall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_Fall);
+}
+
+void BgMoriBigst_Fall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor_MoveForward(&this->dyna.actor);
+    if (this->dyna.actor.posRot.pos.y <= this->dyna.actor.initPosRot.pos.y) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y;
+        BgMoriBigst_SetupLanding(this, globalCtx);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONE_BOUND);
+        func_800800F8(globalCtx, 0x3FC, 8, &this->dyna.actor, 0);
+        func_8002DF38(globalCtx, NULL, 0x3C);
+    }
+}
+
+void BgMoriBigst_SetupLanding(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Camera** cameras = globalCtx->cameraPtrs;
+    s32 quake;
+
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_Landing);
+    this->waitTimer = 18;
+    quake = Quake_Add(cameras[globalCtx->activeCamera], 3);
+    Quake_SetSpeed(quake, 0x61A8);
```

Comment:
```suggestion
    Quake_SetSpeed(quake, 25000);
```
most decompiled usage is round values and decimal so I guess keep it consistent
...
even if it's used like this by `Quake_Calc` lol: `absSpeedDiv = ABS(req->speed) / (f32)0x8000;`